src/tron.jl

```diff
@@ -1,27 +1,72 @@
 #  Some parts of this code were adapted from
 # https://github.com/PythonOptimizers/NLP.py/blob/develop/nlp/optimize/tron.py
 
-export tron
+export tron, TronSolver
 
 tron(nlp::AbstractNLPModel; variant = :Newton, kwargs...) = tron(Val(variant), nlp; kwargs...)
 
 """
-    tron(nlp)
+    tron(nlp; kwargs...)
 
----
+A pure Julia implementation of a trust-region solver for bound-constrained optimization:
+    
+        min f(x)    s.t.    ℓ ≦ x ≦ u
+    
+For an advanced usage, one can first define an `TronSolver` preallocating the memory used in the algorithm and then call `solve!`.
 
     solver = TronSolver(nlp)
-    output = solve!(solver, nlp)
+    solve!(solver, nlp; kwargs...)
+
+# Arguments
+- `nlp::AbstractNLPModel` represents the model solved, see `NLPModels.jl`.
+The keyword arguments may include
+- `subsolver_logger::AbstractLogger = NullLogger()`: subproblem's logger.
+- `x = nlp.meta.x0`: the initial guess.
+- `μ₀::T = T(1e-2)`: algorithm's parameter.
```

Comment:
```suggestion
- `μ₀::T = T(1e-2)`: algorithm parameter.
```
State restrictions on the parameter.